build-support/migration-support/convert_source_to_sources.py

```diff
@@ -0,0 +1,120 @@
+# Copyright 2020 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import argparse
+import tokenize
+from difflib import unified_diff
+from io import BytesIO
+from pathlib import Path
+from token import NAME, OP
+from typing import Dict, List, Optional, Set
+
+
+def main() -> None:
+    args = create_parser().parse_args()
+    build_files: Set[Path] = set(
+        fp
+        for folder in args.folders
+        for fp in [*folder.rglob("BUILD"), *folder.rglob("BUILD.*")]
+        # Check that it really is a BUILD file
+        if fp.is_file() and fp.stem == "BUILD"
+    )
+    updates: Dict[Path, List[str]] = {}
+    for build in build_files:
+        possibly_new_build = maybe_rewrite_build(build)
+        if possibly_new_build is not None:
+            updates[build] = possibly_new_build
+    for build, new_content in updates.items():
+        if args.preview:
+            print(generate_diff(build, new_content))
+        else:
+            build.write_text("\n".join(new_content) + "\n")
+
+
+def create_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="Convert deprecated `source` fields to `sources`.",
+    )
+    parser.add_argument(
+        "folders", type=Path, nargs="+", help="Folders to recursively search for `BUILD` files",
+    )
+    parser.add_argument(
+        "-p",
+        "--preview",
+        action="store_true",
+        help="Output to stdout rather than overwriting BUILD files.",
+    )
+    return parser
+
+
+def maybe_rewrite_line(line: str) -> Optional[str]:
```

Comment:
Would this handle:
```
scala_library(
  source = \
     "messing_with_things.scala"
)
``` 
or 
```
scala_library(
  source = \
     "messing_with_ \
           things.scala"
)
``` ?

It depends on tokenizer and what it does, but if it doesn't, something in the help message to say which cases are not covered would be helpful.

Reply:
It wouldn't attempt to handle this because the tokenizer is line based so isn't capable of next-line look-aheads. Very similar to how grep behaves.

The user would need to manually fix this edge case by grepping and/or relying on Pants to warn them. 

This is why I estimate the fix rate to be 95%-100%. There might be some weird edge cases like this.